compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingInterface.cpp

```diff
@@ -266,15 +278,238 @@ static LogicalResult replaceAllStoresWithTiledVersion(
 
 namespace {
 
-/// Result of the tiled operation.
+/// Result of the tiled operation. The resulted tiled loops may have a different
+/// number of loops of the input loop range. `loops`, `tileSizesForLoops`, and
+/// `boundedSizesForLoops` have the same size.
 struct TilingResult {
   Operation *tiledOp;
-  SmallVector<scf::ForOp> loops;
+  // BitVector for the tiled loops in the loop range
   llvm::SmallBitVector tiledLoops;
+  // offsets for the input loop range
   SmallVector<OpFoldResult> tileOffsets;
+  // tile sizes for the input loop range
   SmallVector<OpFoldResult> tileSizes;
+  // tile sizes requested for the input loop range
+  SmallVector<Value> requestedTileSizes;
+
+  // The result information
+  //
+  // tiled loops.
+  SmallVector<scf::ForOp> loops;
+  // tile sizes for the resulted loops
+  SmallVector<Value> tileSizesForLoops;
+  // bounded tile size for the resulted loops
+  SmallVector<OpFoldResult> boundedSizesForLoops;
+
+  void dump() {
+    llvm::dbgs() << "<TilingResult>\n"
+                 << "tiledOp =\n";
+    if (tiledOp) {
+      tiledOp->dump();
+      llvm::dbgs() << "\n";
+    } else {
+      llvm::dbgs() << "nullptr\n";
+    }
+
+    llvm::dbgs() << "\ntileOffsets:\n";
+    for (auto tileOffset : tileOffsets) {
+      tileOffset.dump();
+    }
+
+    llvm::dbgs() << "\ntileSizes:\n";
+    for (auto tileSize : tileSizes) {
+      tileSize.dump();
+    }
+
+    llvm::dbgs() << "\nrequestedTileSizes:\n";
+    for (auto size : requestedTileSizes) {
+      size.dump();
+    }
+
+    llvm::dbgs() << "# of loops = " << loops.size() << "\n";
+    if (!loops.empty()) {
+      loops[0].dump();
+    }
+
+    llvm::dbgs() << "\ntileSizesForLoops:\n";
+    for (auto size : tileSizesForLoops) {
+      size.dump();
+    }
+
+    llvm::dbgs() << "\nboundedSizesForLoops:\n";
+    for (auto size : boundedSizesForLoops) {
+      size.dump();
+    }
+    llvm::dbgs() << "</TilingResult>\n";
+  }
 };
 
+/// Get the integer tile sizes from the requestedTileSizes
+static FailureOr<SmallVector<int64_t>> getTileSizesAsInt64(
+    ArrayRef<Value> requestedTileSizes) {
+  SmallVector<int64_t> tileSizes;
+
+  for (auto val : requestedTileSizes) {
+    if (auto constIndexOp = val.getDefiningOp<arith::ConstantIndexOp>()) {
+      auto v = constIndexOp.value();
+      // When tiling is not defined, let's use 1 here.
+      tileSizes.push_back(v == 0 ? 1 : v);
```

Comment:
0 usually means no tiling while 1 means tile to size of 1. It seems odd to set 1 here?

Reply:
The value can't be zero for the tiled loops. I will change it. Thanks for catching it.